File: W2D2-Activities2/app/src/main/java/com/example/activities/SecondActivity.java
package com.example.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);


        //İlk sayfadan gelen değeri aldık
        Intent secondIntent = getIntent();
        String url = secondIntent.getStringExtra("urlKey");

        //Alınan değeri ekranda gösterdik
        TextView urlTextView = findViewById(R.id.textViewUrl);
        urlTextView.setText(url);


        //Butonu tanımladık
        Button getWebPage = findViewById(R.id.buttonGetWebPage);
        getWebPage.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                //Butona tıklanınca taracıyı açtık
                Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("http://www."+url));
                startActivity(browserIntent);
            }
        });


        Button buttonShare = findViewById(R.id.buttonShare);
        buttonShare.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                //Intent oluşturduk
                Intent shareIntent = new Intent(Intent.ACTION_SEND);

                //Intente text göndereciğimiz belirttik
                shareIntent.setType("text/plain");

                //Neyi belirttik, yani url
                shareIntent.putExtra(Intent.EXTRA_TEXT, url);

                //Intenti başlattık
                startActivity(Intent.createChooser(shareIntent, "Şununla Paylaş"));

            }
        });


        Button buttonCall = findViewById(R.id.buttonCall);
        buttonCall.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent callIntent = new Intent(Intent.ACTION_CALL , Uri.parse("tel:" + "4440444"));
               startActivity(callIntent);
            }
        });


    }
}

File: W8D2MVVM/app/src/main/java/com/example/w8d1_mvvm/SecondActivity.kt
package com.example.w8d1_mvvm

import android.os.Bundle
import android.widget.TextView
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class SecondActivity : AppCompatActivity() {
    private val viewModel: SecondViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (viewModel.matchResultLiveData.value == null) {
            viewModel.getMatchResults("GS")
        }

        viewModel.matchResultLiveData.observe(this) {
            showMatchResult(it)
        }
    }
    private fun showMatchResult(matchResult: String) {
        val myText = findViewById<TextView>(R.id.my_text)
        myText.text = matchResult
    }
}

class SecondViewModel: ViewModel() {
    val matchResultLiveData = MutableLiveData<String>()

    fun getMatchResults(team: String) {
        val allMatchResults = getAllMatchResults()
        if (team == "GS") {
            val gsMatchResult = allMatchResults[0]
            matchResultLiveData.value = gsMatchResult
        } else if (team == "FB") {
            val fbMatchResult = allMatchResults[1]
            matchResultLiveData.value = fbMatchResult
        }
    }

    private fun getAllMatchResults(): List<String> {
        val matchResults = arrayListOf<String>()
        matchResults.add("GS 1 - 2 GSB")
        matchResults.add("FB 2 - 2 FBB")
        matchResults.add("BJK 1 - 3 BJKB")
        return matchResults
    }
}